Compile error: av_register_all is deprecated

Compilation fails with FFMpeg master since commit https://github.com/FFmpeg/FFmpeg/commit/0694d8702421e7aff1340038559c438b61bb30dd

```
../tests/decodeinputavformat.cpp: In constructor ‘DecodeInputAvFormat::DecodeInputAvFormat()’:
../tests/decodeinputavformat.cpp:32:21: error: ‘void av_register_all()’ is deprecated [-Werror=deprecated-declarations]
     av_register_all();
                     ^
In file included from ../tests/decodeinputavformat.h:22:0,
                 from ../tests/decodeinputavformat.cpp:20:
/home/uaeoff/Work/workspace/media/install/include/libavformat/avformat.h:2024:6: note: declared here
 void av_register_all(void);
      ^~~~~~~~~~~~~~~
../tests/decodeinputavformat.cpp: In constructor ‘DecodeInputAvFormat::DecodeInputAvFormat()’:
../tests/decodeinputavformat.cpp:32:21: error: ‘void av_register_all()’ is deprecated [-Werror=deprecated-declarations]
     av_register_all();
                     ^
In file included from ../tests/decodeinputavformat.h:22:0,
                 from ../tests/decodeinputavformat.cpp:20:
/home/uaeoff/Work/workspace/media/install/include/libavformat/avformat.h:2024:6: note: declared here
 void av_register_all(void);
      ^~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [Makefile:667: ../tests/grid-decodeinputavformat.o] Error 1
make[2]: *** Waiting for unfinished jobs....
cc1plus: all warnings being treated as errors
make[2]: *** [Makefile:615: ../tests/decodeinputavformat.o] Error 1
```

--- a/tests/decodeinputavformat.cpp
+++ b/tests/decodeinputavformat.cpp
@@ -29,7 +29,9 @@
 DecodeInputAvFormat::DecodeInputAvFormat()
 :m_format(NULL),m_videoId(-1), m_codecId(AV_CODEC_ID_NONE), m_isEos(true)
 {
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(58, 9, 100)
     av_register_all();
+#endif
 
     av_init_packet(&m_packet);
 }

